### PR TITLE
Add discord link

### PR DESCRIPTION
**Pull request by marsaldev**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Adding the field to add a discord link, request from @Petruzzo73 in this issue: https://github.com/PrestaShop/PrestaShop/issues/16356#issuecomment-1046090817
| Type?         | improvement 
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | [Fixes PrestaShop/Prestashop#12.](https://github.com/PrestaShop/PrestaShop/issues/16356)
| How to test?  | #

~~I think we should add the discord icon, like I've read in the conversation of this PR (@PierreRambaud, @TristanLDD): https://github.com/PrestaShop/ps_socialfollow/pull/15~~
Added icon at module level, through a css file. The icon can be overridden with a css file and icon at theme level

### CODE REVIEW

**Comment by matthieu-rolland** [comment]
I think you put the wrong issue in your table 🤔

**Comment by marsaldev** [comment]
> I think you put the wrong issue in your table 🤔

Actually there isn't a real related issue 😊 but only the mentioned comment from @Petruzzo73

**Review by NeOMakinG — APPROVED** [review]
The module isn't that flexible, you could be able to add any social network you want if you provide the icon, SVG or whatever

**Comment by marsaldev** [comment]
> The module isn't that flexible, you could be able to add any social network you want if you provide the icon, SVG or whatever

Agree with you, a big change should be made, remove all that separated fields and create its own table where record each 'social' and associate link and icon/image

File: ps_socialfollow.php [hunk 1]
```diff
@@ -323,6 +333,14 @@ public function getWidgetVariables($hookName = null, array $configuration = [])
                 'url' => $sf_linkedin,
             ];
         }
+
```
Comment (kpodemski) [inline ps_socialfollow.php:336]:
```suggestion

```

**Comment by marsaldev** [comment]
> Have you checked if the Discord icon is available in front office?

As far as I know there no discord icon in the classic theme, how we can solve this?

**Comment by kpodemski** [comment]
@marsaldev 

it has to be implemented on the theme level, too. You should make a pull request in the classic theme repository to add a .svg icon :-)

@NeOMakinG, what about the new theme?

**Comment by NeOMakinG** [comment]
> @marsaldev
> 
> it has to be implemented on the theme level, too. You should make a pull request in the classic theme repository to add a .svg icon :-)
> 
> @NeOMakinG, what about the new theme?

Nice catch btw

The new theme should probably have the icon too, so it needs 2 more PRs, but it's a nightmare, why icons aren't on module level? Some access right issues? 

- Module should use icons on module level
- If it's overridden in the theme, it will use the override

If we are able to do this, this is way more maintainable than the actual behavior

**Comment by marsaldev** [comment]
> @NeOMakinG 
> The new theme should probably have the icon too, so it needs 2 more PRs, but it's a nightmare, why icons aren't on module level? Some access right issues?
> 
> * Module should use icons on module level
> * If it's overridden in the theme, it will use the override
> 
> If we are able to do this, this is way more maintainable than the actual behavior

We can start a "transition" with this one, I can create a PR to add an icon only for Discord

**Comment by NeOMakinG** [comment]
wdyt @PrestaShop/prestashop-maintainers , maybe I miss something

**Comment by PierreRambaud** [comment]
Icons are related to the module, so it should be in the module first, then if the file is found in the current theme, it can be used.
But I see no reason to store icons only in the theme.

**Review by saulaski — DISMISSED** [review]

File: views/css/ps_socialfollow.css [hunk 2]
```diff
@@ -0,0 +1,5 @@
+.discord {
```
Comment (kpodemski) [inline views/css/ps_socialfollow.css:1]: @marsaldev this class is too generic, let's go with the, maybe:

```suggestion
.ps-socialfollow-discord {
```

cc @NeOMakinG

Comment (marsaldev) [reply views/css/ps_socialfollow.css:1]: I thought about it 😅 but I have aligned the name to old classes.
But I agree with you 👍🏻

Comment (NeOMakinG) [reply views/css/ps_socialfollow.css:1]: Agree

**Review by NeOMakinG — APPROVED** [review]

**Comment by saulaski** [comment]
That's a lot of commits, please rebase your PR :)

File: assets/icon/Discord-Logo-Color.svg [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<svg width="71" height="55" viewBox="0 0 71 55" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (Progi1984) [inline assets/icon/Discord-Logo-Color.svg:1]: Please rename the icon to `discord`.svg ?

**Review by Progi1984 — CHANGES_REQUESTED** [review]
@marsaldev Hi, thank you for your contribution.

Where did you find the icon ?
What is the license of this image ? (@ttoine)

Thanks for your answer.

**Comment by marsaldev** [comment]
> @marsaldev Hi, thank you for your contribution.
> 
> Where did you find the icon ? What is the license of this image ? (@ttoine)
> 
> Thanks for your answer.

Taken on the official discord web site: https://discord.com/branding
I don't see any mention to any licence on the page 😕

File: assets/icon/Discord-Logo-Color.svg (hunk 3 above)
Comment (marsaldev) [reply assets/icon/Discord-Logo-Color.svg]: Done 😉

**Comment by saulaski** [comment]
Quote form https://discord.com/branding
> We’re a playful and fun brand that doesn't take ourselves too seriously. That said, we're keen on keeping things quality. If you're gonna use our art for something please keep it tasteful and send it our way for approval.
Does it mean any usage of the logo, commercial or not, is to be approved by Discord ?

**Review by kpodemski — DISMISSED** [review]

**Review by saulaski — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Comment by florine2623** [comment]
ping @PrestaShop/product-team ,

The issue related to this PR is closed and doesn't involve a discord link, but other social links.
What should be done ? Re-open the issue ? Create another one ? Something else ?

Thanks!

**Comment by fabienvallon** [comment]
We can re-open.

@ttoine could you tell us about the icon license

**Comment by ttoine** [comment]
My point of view:
Using an icon pack with a creative commons by-sa licence (or equivalent, or more permissive) would simplify the license problem: the IP work is done by the creator of the pack, and we don't have to check each time we add a new social network. We just have to update the pack from time to time or even better, to use it as a dependency at build. (And yes it would be great, then, to provide a way to add unlimited custom social network, of course in an other pr/issue, with icons from the pack or custom icons provided by the users --PM team can work on that)
Also in term of design, it's coherent, instead of having each icon being very different. And it solves the problem of future social networks all at once.

About the license of one specific logo: 
It's a company/product/service logo, so it's a brand. In this case it's owned by Discord. If it's taken from their branding guidelines pack of artwork, then their copyright applies (like PrestaShop logo or any company/product/service logo, btw). We must add something about it somewhere, maybe in the readme.md, and if it was not done previously, it must be done too for other logos distributed with the module (same for the classic theme if any)
Something like the list of all the logos that are shipped, with a link to where people can find the original, and the copyright information

**Review by florine2623 — CHANGES_REQUESTED** [review]
@marsaldev , thanks for your PR !

As @ttoine said
>We must add something about it somewhere, maybe in the readme.md

@ttoine could you provide what exactly should be said ? so @marsaldev can add it to the readme ?

**Comment by marsaldev** [comment]
ping @ttoine Any help? 😊

**Comment by ttoine** [comment]
at the bottom of the readme, you can add a section called "Third party logos" and add a list of icons and a link to the page where people can find more information. (typically for Discord the page where you downloaded the icon)

**Review by NeOMakinG — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

**Review by kpodemski — APPROVED** [review]

**Review by Sinepel — DISMISSED** [review]

**Comment by florine2623** [comment]
Hello @marsaldev ,

Thanks for the PR !

The Discord logo doesn't display on my shop, the logo is present in /assets/icon/discord.svg  : 

https://user-images.githubusercontent.com/16019289/161512677-39c09cf2-2331-4e80-b2cd-a7657ed8fae0.mov

Could you check ? The rest looks good. The BO configuration for the logo is good too.

**Comment by marsaldev** [comment]
@florine2623 
> Hello @marsaldev ,
> 
> Thanks for the PR !
> 
> The Discord logo doesn't display on my shop, the logo is present in /assets/icon/discord.svg :
> 
>  Screen.Recording.2022-04-04.at.11.10.43.mov 
> Could you check ? The rest looks good. The BO configuration for the logo is good too.

Weird 🤔, have you hooked the module in `actionFrontControllerSetMedia`?
Have you checked if the ps_socialfollow.css is loaded properly?

**Comment by kpodemski** [comment]
@marsaldev I think you forgot to create an upgrade file for the module to use this new hook in the new version

**Comment by marsaldev** [comment]
> @marsaldev I think you forgot to create an upgrade file for the module to use this new hook in the new version

Yep, just done 😉

File: upgrade/upgrade-2.3.0.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+if (!defined('_PS_VERSION_')) {
+    exit;
+}
+
+function upgrade_module_2_3_0($object)
+{
+    return (bool) $object->registerHook('actionFrontControllerSetMedia');
```
Comment (atomiix) [inline upgrade/upgrade-2.3.0.php:32]:
```suggestion
    return (bool) $object->registerHook('actionFrontControllerSetMedia') && Configuration::updateValue('BLOCKSOCIAL_DISCORD', '');
```

**Review by atomiix — APPROVED** [review]

**Comment by florine2623** [comment]
@marsaldev ,

Now I have the logo but the color is not correct and it seems a little too bug :

https://user-images.githubusercontent.com/16019289/161725026-0e325475-d5ab-44db-9e9e-87f2427b803b.mov

Could you check ? ^^
Thanks!

**Comment by marsaldev** [comment]
Hi @florine2623,
you are right, I've used the coloured one because I thought we cannot use others colours, but discord provides the white and black too.
So I've update the icon with the white one and resized a little bit the icon, now I think it shows properly 😊

**Review by florine2623 — APPROVED** [review]
Hello @marsaldev ,

Beautiful ! Thanks it is QA ✅ 

https://user-images.githubusercontent.com/16019289/161947519-46c7188f-0ecb-4717-9b0e-263a3c4e7ceb.mov

**Comment by kpodemski** [comment]
Thank you @marsaldev ❤️

**Comment by matks** [comment]
Bravo @marsaldev

**Comment by SharakPL** [comment]
This is wrong! Only discord icon has styles provided by the module and all the rest is styled in the theme. **It shouldn't be like that!**. Plus the classic theme loads icons as background of `a::before` while css added by this PR forces discord as a background of `a`. It messes up custom themes:

obraz

All svg icons should be loaded by the template, not by css. Just like it was done in Hummingbird, which I've just noticed has discord and tiktok still missing:
https://github.com/PrestaShop/hummingbird/blob/34e15d0bd93cfeb2c69e407e3d89a91eec0559cd/modules/ps_socialfollow/ps_socialfollow.tpl#L9-L29

**Comment by SharakPL** [comment]
Sure thing. I just need to finish my current project first...

**Pull request merged**